File: go/3147. Taking Maximum Energy From the Mystic Dungeon.go
func maximumEnergy(energy []int, k int) int {
	answer := math.MinInt32
	for i := len(energy) - 1; i > len(energy)-1-k; i-- {
		currentSum := 0
		for j := i; j >= 0; j -= k {
			currentSum += energy[j]
			answer = max(answer, currentSum)
		}
	}
	return answer
}

File: java/3147. Taking Maximum Energy From the Mystic Dungeon.java
class Solution {
    public int maximumEnergy(int[] energy, int k) {
        int answer = Integer.MIN_VALUE;
        for (int i = energy.length-1; i > energy.length-1-k; i--) {
            int currentSum = 0;
            for (int j = i; j >= 0; j -= k) {
                currentSum += energy[j];
                answer = Math.max(answer, currentSum);
            }
        }
        return answer;
    }
}